diesel/src/types/impls/date_and_time/chrono.rs

```diff
@@ -45,6 +47,38 @@ impl ToSql<Timestamp> for NaiveDateTime {
     }
 }
 
+fn midnight() -> NaiveTime {
+    NaiveTime::from_hms(0, 0, 0)
+}
+
+impl ToSql<Time> for NaiveTime {
+    fn to_sql<W: Write>(&self, out: &mut W) -> Result<IsNull, Box<Error>> {
+        let duration = *self - midnight();
+        match duration.num_microseconds() {
+            Some(offset) => ToSql::<Time>::to_sql(&PgTime(offset), out),
+            None => {
+                let error_message = format!("{:?} as microseconds is too large to fit in an i64",
```

Comment:
I think the `None` case is just `unreachable!()`, since `NaiveTime` cannot represent more than 24 hours as microseconds.